test/rules/src/https_everywhere_checker/rules.py

```diff
@@ -180,15 +180,55 @@ def _determineTestApplication(self):
 		return self.test_application_problems
 
 	def getCoverageProblems(self):
-		"""Verify that each rule and each exclusion has the right number of tests
-			 that applies to it. TODO: Also check that each target has the right
+		"""Verify that each target, rule and exclusion has the right number of tests
+			 that applies to it. Also check that each target has the right
 			 number of tests. In particular left-wildcard targets should have at least
 			 three tests. Right-wildcard targets should have at least ten tests.
 
 			 Returns an array of strings reporting any coverage problems if they exist,
 			 or empty list if coverage is sufficient.
 			 """
 		problems = self._determineTestApplication()
+
+		# First, check each target has the right number of tests
+		myTestTargets = []
+
+		# Only take tests which are not excluded into account
+		for test in self.tests:
+			if not self.excludes(test.url):
+				urlParts = urlparse(test.url)
+				hostname = urlParts.hostname
+				myTestTargets.append(hostname)
+
+		for target in self.targets:
+			actual_count = 0
+			needed_count = 1
+
+			if target.startswith("*."):
+				needed_count = 3
+
+			if target.endswith(".*"):
+				needed_count = 10
+
+			# non-wildcard target always have a implicit test url, if is it not excluded
+			if not "*" in target and not self.excludes(("http://%s/" % target)):
+				continue
+
+			# '*.example.com' match 'www.example.com' but not 'secure.account.exmple.com'
```

Comment:
@Hainish does `*.example.com` matches `www.example.com`, `secure.account.example.com` and `example.com`? I am a bit confused seeing https://support.google.com/customsearch/answer/71826?hl=en